src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-list/rbd-list.component.ts

```diff
@@ -0,0 +1,256 @@
+      .subscribe(
+      (resp: any[]) => {
+        let images = [];
+        const viewCacheStatusMap = {};
+        resp.forEach(pool => {
+          if (_.isUndefined(viewCacheStatusMap[pool.status])) {
+            viewCacheStatusMap[pool.status] = [];
+          }
+          viewCacheStatusMap[pool.status].push(pool.pool_name);
+          images = images.concat(pool.value);
+        });
+        const viewCacheStatusList = [];
+        _.forEach(viewCacheStatusMap, (value, key) => {
+          viewCacheStatusList.push({
+            status: parseInt(key, 10),
+            statusFor: (value.length > 1 ? 'pools ' : 'pool ') +
+            '<strong>' + value.join('</strong>, <strong>') + '</strong>'
+          });
+        });
+        this.viewCacheStatusList = viewCacheStatusList;
+        images.forEach(image => {
+          image.executingTasks = this._getExecutingTasks(executingTasks,
+            image.pool_name, image.name);
+        });
+        this.images = this.merge(images, executingTasks);
+        this.executingTasks = executingTasks;
+      },
+      () => {
+        this.viewCacheStatusList = [{status: ViewCacheStatus.ValueException}];
+      }
+    );
+  }
+
+  _getExecutingTasks(executingTasks: ExecutingTask[], poolName, imageName): ExecutingTask[] {
+    const result: ExecutingTask[] = [];
+    executingTasks.forEach(executingTask => {
+      if (executingTask.name === 'rbd/snap/create' ||
+        executingTask.name === 'rbd/snap/delete' ||
+        executingTask.name === 'rbd/snap/edit' ||
+        executingTask.name === 'rbd/snap/rollback') {
+        if (poolName === executingTask.metadata['pool_name'] &&
+          imageName === executingTask.metadata['image_name']) {
+          result.push(executingTask);
+        }
+      }
+    });
+    return result;
+  }
+
+  private merge(rbds: RbdModel[], executingTasks: ExecutingTask[] = []) {
+    const resultRBDs = _.clone(rbds);
+    executingTasks.forEach((executingTask) => {
+      if (executingTask.name === 'rbd/create') {
+        const exists = resultRBDs.some((rbd) => {
+          return rbd.pool_name === executingTask.metadata['pool_name'] &&
+            rbd.name === executingTask.metadata['image_name'];
+        });
+        if (!exists) {
+          const rbdModel = new RbdModel();
+          rbdModel.name = executingTask.metadata['image_name'];
+          rbdModel.pool_name = executingTask.metadata['pool_name'];
+          rbdModel.cdExecuting = 'creating';
+          resultRBDs.push(rbdModel);
+        }
+      } else if (executingTask.name === 'rbd/delete') {
+        const rbdToDelete = resultRBDs.find((rbd) => {
+          return rbd.pool_name === executingTask.metadata['pool_name'] &&
+            rbd.name === executingTask.metadata['image_name'];
+        });
+        if (rbdToDelete) {
+          rbdToDelete.cdExecuting = 'deleting';
+        }
+      } else if (executingTask.name === 'rbd/edit') {
+        const rbdToDelete = resultRBDs.find((rbd) => {
```

Comment:
(Sorry didn't spot this before)

Could refactor it to something that is similar to [this](https://github.com/ceph/ceph/pull/21385/files#diff-3218aa223ce1c378579e8578ec8e1429R113)?
Also the const `rbdToDelete` is very irritating for the editing part.

Reply:
Done